crates/ra_syntax/src/ast/expr_extensions.rs

```diff
@@ -305,44 +305,34 @@ impl ast::Literal {
             .unwrap()
     }
 
-    pub fn kind(&self) -> LiteralKind {
-        match self.token().kind() {
-            INT_NUMBER => {
-                let int_suffix_list = [
-                    "isize", "i128", "i64", "i32", "i16", "i8", "usize", "u128", "u64", "u32",
-                    "u16", "u8",
-                ];
+    fn find_suffix(text: &str, possible_suffixes: &[&str]) -> Option<SmolStr> {
+        possible_suffixes
+            .iter()
+            .find(|&suffix| text.ends_with(suffix))
+            .map(|&suffix| SmolStr::new(suffix))
+    }
 
-                // The lexer treats e.g. `1f64` as an integer literal. See
```

Comment:
I think this comment is useful. Without the comment it’s not necessarily obvious why we’d check float suffixes for a token of kind INT_NUMBER (and indeed the code incorrectly didn’t do this before the PR in which the comment was added).

Reply:
Nit (in case you touch this code again): "below".